### DISCUSSION TITLE
Create API for TLS version Restriction

### DISCUSSION THREAD

**Discussion by CCSrl** [Azure/azure-iot-sdk-c#2003] [Ideas]
Regarding to current version of azure-iot-sdk-c, there's no Openssl function like SSL_CTX_set_min_proto_version(SSL_CTX *ctx, int version) which able to limit the TLS version as shown as image below.
https://www.openssl.org/docs/man1.1.1/man3/SSL_CTX_set_max_proto_version.html
![TLS_version_filter](https://user-images.githubusercontent.com/78889420/122171457-5164e780-ceb2-11eb-9ac8-34df0eb5eda8.png)

Is it possible to create a function in create_openssl_instance(TLS_IO_INSTANCE* tlsInstance) to enable to limit the version supported protocol according to the version set in tlsio_openssl_setoption?
https://github.com/Azure/azure-c-shared-utility/blob/master/adapters/tlsio_openssl.c#L1049
https://github.com/Azure/azure-c-shared-utility/blob/master/adapters/tlsio_openssl.c#L1617

**Comment by ericwolz** [comment]
Thanks, we will consider this for a future enhancement.
